File: 2996-smallest-missing-integer-greater-than-sequential-prefix-sum/smallest-missing-integer-greater-than-sequential-prefix-sum.py
class Solution(object):
    def missingInteger(self, nums):
        n =len(nums)
        my_set =set(nums)
        sum =nums[0]
        for i in range(1,n):
            if nums[i]==nums[i-1]+1:
                sum+=nums[i]
            else:
                break
        while sum in my_set:
            sum+=1
        
        return sum
        """
        :type nums: List[int]
        :rtype: int
        """
        

File: 2996-smallest-missing-integer-greater-than-sequential-prefix-sum/smallest-missing-integer-greater-than-sequential-prefix-sum.ts
function missingInteger(nums: number[]): number {
    const n = nums.length;
    const st:Set<number>=new Set(nums);
    let sum =nums[0];
    for(let i=1;i<n;i++){
        if(nums[i]===nums[i-1]+1){
            sum+=nums[i];
        }
        else{
            break;
        }
    }
    while(st.has(sum)){
        sum++;
    }
    return sum;
};